protostar/starknet_gateway/gateway_facade.py

```diff
@@ -281,3 +284,44 @@ def __init__(self, function_name: str):
 class ContractNotFoundException(ProtostarException):
     def __init__(self, contract_address: AddressRepresentation):
         super().__init__(f"Tried to call unknown contract:\n{contract_address}")
+
+
+class InvalidInputException(ProtostarException):
+    pass
+
+
+def validate_deploy_input(compiled_contract_json: str, inputs: List[int]) -> None:
```

Comment:
can you move logic of json parsing out of this function?

Reply:
we should probably try-catch it too

Reply:
done

Reply:
> we should probably try-catch it too

on the other hand I don't think checking for invalid input is a job of this module, wdyt?

Reply:
```suggestion
def validate_deploy_input(abi: AbiType, inputs: List[int]) -> None:
```

This is the interface I would expect here. Currently, this function is mixing concerns: parsing and contract input validation (actually the latter could be divided into two stages -- finding the constructor and then validating its inputs, but this is less important)